Sources/BigIntModule/BigInt.swift

```diff
@@ -9,6 +9,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Foundation
```

Comment:
Numerics can't have a dependency on Foundation.

Reply:
I needed the log() function to do a quick digit count.
I'll see if this can be done another way.

Reply:
Violet does it differently (without `log`). It uses magic numbers from V8 (google javascrtipt thinge). It is a really good approximation for the upper bound.

Unfortunately it is an upper bound, so sometimes it has a tiny overallocation. This is why after the filling of the buffer there is a `if index != -1` check (with comment 'Check for invalid capacity estimation'). If we overallocate then we have to move the whole result (because we are filling from the back). Overallocation is not a problem (because it is tiny), the shift may be because of performance.

Btw. just so we are clear: I confirm that for radix 10 (most common case) this overallocation happens, which means that sometime we have to shift. But even with this penalty the performance is much better than the current `main` branch. 

Btw2. There is a way to avoid this shifting. In Violet if your platform does not have `String(unsafeUninitializedCapacity: capacity, initializingUTF8With: initializer)` then we implement our won shim. At the end of this shim we call `String(cString: ptr) // Borrows 'ptr' to create owned copy.`. To avoid shifting we could just increment the `ptr` so that the buffer starts exactly  where the digits start. Thought, obviously if you platform has ``String(unsafeUninitializedCapacity: capacity, initializingUTF8With: initializer)`` then you have to correctly fill the provided buffer. Anyway it does not really matter...

Reply:
Thanks @LiarPrincess for all your comments.  Hope you get better soon.
@stephentyrone, I've removed the Foundation dependency by using a table.